From 1cb1d9aa3f6660db9ec18185b5202fb2f91d5d60 Mon Sep 17 00:00:00 2001
From: Edward Welbourne <eddy@chaos.org.uk>
Date: Mon, 25 Jan 2010 17:32:33 +0100
Subject: [PATCH 01/02] libfstt: Fix segfault on font file open error

RandomAccessFile::RandomAccessFile(char *filename)'s error path, on
failure to open a file, neglects to set absbase; this leads to
openError() failing to report that open failed, so instances think
they're OK and try to use methods that should only be used if open
succeeded, with consequent segfaults.
---
 libfstt/rafile.cc |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/libfstt/rafile.cc b/libfstt/rafile.cc
index 6afac75..7155fcb 100644
--- a/libfstt/rafile.cc
+++ b/libfstt/rafile.cc
@@ -176,7 +176,7 @@ RandomAccessFile::RandomAccessFile(char
 	int fd = open(fileName, O_RDONLY);
 	if (fd <= 0) {
 		debug("Cannot open \"%s\"\n", fileName);
-		ptr = base = 0;
+		ptr = absbase = base = 0;
 		return;
 	}
 	struct stat st;
-- 
1.7.1

